libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveComponentRegistration.cs

```diff
@@ -19,163 +16,165 @@
-            yield return new TypeRegistration<MultiLanguageGenerator>(MultiLanguageGenerator.DeclarativeType);
-            yield return new TypeRegistration<TemplateEngineLanguageGenerator>(TemplateEngineLanguageGenerator.DeclarativeType);
+            yield return new DeclarativeType<ResourceMultiLanguageGenerator>(ResourceMultiLanguageGenerator.DeclarativeType);
+            yield return new DeclarativeType<MultiLanguageGenerator>(MultiLanguageGenerator.DeclarativeType);
+            yield return new DeclarativeType<TemplateEngineLanguageGenerator>(TemplateEngineLanguageGenerator.DeclarativeType);
 
             // Dialogs
-            yield return new TypeRegistration<AdaptiveDialog>(AdaptiveDialog.DeclarativeType);
+            yield return new DeclarativeType<AdaptiveDialog>(AdaptiveDialog.DeclarativeType);
 
             // test actions
-            yield return new TypeRegistration<AssertCondition>(AssertCondition.DeclarativeType);
-            yield return new TypeRegistration<TestScript>(TestScript.DeclarativeType);
-            yield return new TypeRegistration<UserSays>(UserSays.DeclarativeType);
-            yield return new TypeRegistration<UserTyping>(UserTyping.DeclarativeType);
-            yield return new TypeRegistration<UserConversationUpdate>(UserConversationUpdate.DeclarativeType);
-            yield return new TypeRegistration<UserActivity>(UserActivity.DeclarativeType);
-            yield return new TypeRegistration<UserDelay>(UserDelay.DeclarativeType);
-            yield return new TypeRegistration<AssertReply>(AssertReply.DeclarativeType);
-            yield return new TypeRegistration<AssertReplyOneOf>(AssertReplyOneOf.DeclarativeType);
-            yield return new TypeRegistration<AssertReplyActivity>(AssertReplyActivity.DeclarativeType);
+            yield return new DeclarativeType<AssertCondition>(AssertCondition.DeclarativeType);
+            yield return new DeclarativeType<TestScript>(TestScript.DeclarativeType);
+            yield return new DeclarativeType<UserSays>(UserSays.DeclarativeType);
+            yield return new DeclarativeType<UserTyping>(UserTyping.DeclarativeType);
+            yield return new DeclarativeType<UserConversationUpdate>(UserConversationUpdate.DeclarativeType);
+            yield return new DeclarativeType<UserActivity>(UserActivity.DeclarativeType);
+            yield return new DeclarativeType<UserDelay>(UserDelay.DeclarativeType);
+            yield return new DeclarativeType<AssertReply>(AssertReply.DeclarativeType);
+            yield return new DeclarativeType<AssertReplyOneOf>(AssertReplyOneOf.DeclarativeType);
+            yield return new DeclarativeType<AssertReplyActivity>(AssertReplyActivity.DeclarativeType);
         }
 
-        public override IEnumerable<JsonConverter> GetConverters(ISourceMap sourceMap, IRefResolver refResolver, Stack<string> paths)
```

Comment:
Is the sourcemap stuff embedded in some other way?  Does the debugger still work?

Reply:
SourceMap is simply a static.  Debugger opens a port for debugging, and all sources in the process are accessed through that single port, so no need for instances of source maps.  DebugSupport.SourceMap.

---
In reply to: [376540371](https://github.com/microsoft/botbuilder-dotnet/pull/3350#discussion_r376540371) [](ancestors = 376540371)